tests/nativeapp/codegen/snowpark/test_python_processor.py

```diff
@@ -25,6 +28,56 @@
 
 PROJECT_ROOT = Path("/path/to/project")
 
+
+# contextlib.chdir isn't available before Python 3.11, so this is an alternative for older versions
+@contextlib.contextmanager
+def in_cwd(path: Path) -> Iterator[None]:
+    old_cwd = os.getcwd()
+    os.chdir(str(path))
+    try:
+        yield
+    except Exception:
+        pass
+
+    os.chdir(old_cwd)
+
+
+def stringify(p: Path):
+    if p.is_dir():
+        return f"d {p}"
+    else:
+        return f"f {p}"
+
+
+def all_paths_under_dir(root: Path) -> List[Path]:
+    assert root.is_dir()
+
+    paths: Set[Path] = set()
+    for subdir, dirs, files in os.walk(root):
```

Comment:
`os.walk` does not have a guaranteed order, so we should sort `dirs` and `files` to ensure the output is stable.

Reply:
Absolutely, that happens below, the entire output list is sorted.

Reply:
Lol 🤦